verifier/verifier.go

```diff
@@ -63,6 +67,24 @@ func New(trustPolicy *trustpolicy.Document, trustStore truststore.X509TrustStore
 	}, nil
 }
 
+// NewWithRevocationClient creates a new verifier given trustPolicy, trustStore, pluginManager, and revocationClient
+func NewWithRevocationClient(trustPolicy *trustpolicy.Document, trustStore truststore.X509TrustStore, pluginManager plugin.Manager, revocationClient *http.Client) (notation.Verifier, error) {
+	if revocationClient == nil {
+		revocationClient = &http.Client{Timeout: 5 * time.Second}
+	}
+	result, err := New(trustPolicy, trustStore, pluginManager)
+	if err != nil {
+		return nil, err
+	}
+	if v, ok := result.(*verifier); !ok {
+		return result, nil
+	} else {
+		v.revocationClient = revocationClient
+		return v, nil
+	}
```

Comment:
i'm confused, why is this type check needed?

Reply:
The New function return type is notation.Verifier rather than the verifier struct that we know it should be. Thus, a cast is needed. It should never actually fail though

Reply:
Can we get rid of the if-else block since it is causing confusion and simply assign the `revocationClient` to `v` and return. It is fine to check `ok` field and fail (though it never happens ideally)